packages/block-editor/src/components/inserter/menu.js

```diff
@@ -60,60 +60,100 @@ function InserterMenu( {
 
 	const showPatterns = ! destinationRootClientId && hasPatterns;
 
-	const onInsert = ( blocks ) => {
-		onInsertBlocks( blocks );
-		onSelect();
-	};
+	const onInsert = useCallback(
+		( blocks ) => {
+			onInsertBlocks( blocks );
+			onSelect();
+		},
+		[ onInsertBlocks, onSelect ]
+	);
 
-	const onInsertPattern = ( blocks, patternName ) => {
-		onInsertBlocks( blocks, { patternName } );
-		onSelect();
-	};
+	const onInsertPattern = useCallback(
+		( blocks, patternName ) => {
+			onInsertBlocks( blocks, { patternName } );
+			onSelect();
+		},
+		[ onInsertBlocks, onSelect ]
+	);
 
-	const onHover = ( item ) => {
-		onToggleInsertionPoint( !! item );
-		setHoveredItem( item );
-	};
+	const onHover = useCallback(
+		( item ) => {
+			onToggleInsertionPoint( !! item );
+			setHoveredItem( item );
+		},
+		[ onToggleInsertionPoint, setHoveredItem ]
+	);
 
-	const onClickPatternCategory = ( patternCategory ) => {
-		setSelectedPatternCategory( patternCategory );
-	};
+	const onClickPatternCategory = useCallback(
+		( patternCategory ) => {
+			setSelectedPatternCategory( patternCategory );
+		},
+		[ setSelectedPatternCategory ]
+	);
 
-	const blocksTab = (
-		<>
-			<div className="block-editor-inserter__block-list">
-				<BlockTypesTab
-					rootClientId={ destinationRootClientId }
-					onInsert={ onInsert }
-					onHover={ onHover }
-					showMostUsedBlocks={ showMostUsedBlocks }
-				/>
-			</div>
-			{ showInserterHelpPanel && (
-				<div className="block-editor-inserter__tips">
-					<VisuallyHidden as="h2">
-						{ __( 'A tip for using the block editor' ) }
-					</VisuallyHidden>
-					<Tips />
+	const blocksTab = useMemo(
```

Comment:
I believe memoizing elements or render props is useless

Reply:
These elements are used inside a render function. In this case, the render function will return a totally new element every time it's called - which results in all items being rerendered. If we are using memorization then it will reuse the same element every time.

Reply:
React already does diffing when it comes to element, so I'm not sure that's necessary. Also since this is more top level, it's less necessary than memorizing leafs. Do you think we can try removing the memoization and see what impact it has on the numbers?

Reply:
Before:
```
│  inserterHover   │                '38.89 ms'                │ '69.18 ms'  │
│ minInserterHover │                '35.32 ms'                │ '64.11 ms'  │
│ maxInserterHover │                '48.13 ms'                │ '76.42 ms'  │
```

Reply:
Interesting, so all the performance gain is coming from the memoization of the tabs?

Reply:
If you boot up React profiler and enable "highlight updates" then you can clearly see all the items are rerendering when the memo is removed.

Since the passed children is a function, it behaves differently than just a simple element. That's why it's creating a whole new tree without on every render without the memo.

Reply:
Ok, I think I still need to understand why the top component is rerendering at all but let's bring back the memorization for now. Thanks for your work and explorations. Highly appreciated and sorry for the back and forth.

Reply:
No worries! My thanks for going on with this!